src/test/unit/CMakeLists.txt

```
@@ -0,0 +1,12 @@
+cmake_minimum_required(VERSION 3.12)
```

```suggestion
cmake_minimum_required(VERSION 3.15)
```
Make this 3.15 to match `src/CMakeLists.txt`?
Optional, as I suppose the "minimum" for this particular script is 3.12, but on the assumption we are building on the same environment as main we need 3.15 there.